osu.Game.Rulesets.Mania/Objects/Drawables/DrawableManiaHitObject.cs

```diff
@@ -23,10 +23,6 @@ public abstract class DrawableManiaHitObject : DrawableHitObject<ManiaHitObject>
 
         protected readonly IBindable<ScrollingDirection> Direction = new Bindable<ScrollingDirection>();
 
-        // Leaving the default (10s) makes hitobjects not appear, as this offset is used for the initial state transforms.
-        // Calculated as DrawableManiaRuleset.MAX_TIME_RANGE + some additional allowance for velocity < 1.
-        protected override double InitialLifetimeOffset => 30000;
```

Comment:
I think this will provide quite a good performance boost in mania due to less notes being alive.